src/Symfony/Bundle/FrameworkBundle/Routing/RouteCollectionBuilder.php

```diff
@@ -0,0 +1,465 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Routing;
+
+use Symfony\Component\Config\Exception\FileLoaderLoadException;
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\Config\Resource\ResourceInterface;
+use Symfony\Component\Routing\RouteCollection;
+use Symfony\Component\Routing\Route;
+
+/**
+ * Helps add and import routes into a RouteCollection.
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+class RouteCollectionBuilder
+{
+    private $loader;
+
+    /**
+     * A mixture of different objects that hold routes.
+     *
+     * @var Route[]|RouteCollectionBuilder[]
+     */
+    private $routes = array();
+
+    private $defaults = array();
+
+    private $prefix;
+
+    private $host;
+
+    private $condition;
+
+    private $requirements = array();
+
+    private $options = array();
+
+    private $schemes;
+
+    private $methods;
+
+    private $resources = array();
+
+    private $controllerClass;
+
+    /**
+     * @param LoaderInterface $loader
+     */
+    public function __construct(LoaderInterface $loader)
+    {
+        $this->loader = $loader;
+    }
+
+    /**
+     * Import an external routing resource, like a file.
+     *
+     * Returns a RouteCollectionBuilder so you can continue to tweak options on the routes.
+     *
+     * @param mixed  $resource
+     * @param string $prefix
+     * @param string $type
+     *
+     * @return RouteCollectionBuilder
+     */
+    public function import($resource, $prefix = null, $type = null)
+    {
+        /** @var RouteCollection $subCollection */
+        $subCollection = $this->resolve($resource, $type)->load($resource, $type);
```

Comment:
Why not have a private `load()` method that loads the resource; the chain call looks weird to me even more that you need to pass the same arguments to both.